src/d3d8/d3d8_device.cpp

```diff
@@ -90,9 +91,13 @@ namespace dxvk {
         return E_FAIL;
       
       case D3DDEVINFOID_VCACHE:
-        // Docs say response should be S_FALSE, but we'll let D9VK
-        // decide based on the value of supportVCache. D3DX8 calls this.
+        // The query will return D3D_OK on Nvidia and D3DERR_NOTAVAILABLE on AMD/Intel
+        // in D3D9, however in the case of the latter we'll need to return a
+        // zeroed out query result and S_FALSE. This behavior has been observed both
+        // on modern native AMD drivers and D3D8-era native ATI drivers.
         res = GetD3D9()->CreateQuery(d3d9::D3DQUERYTYPE_VCACHE, &pQuery);
+        if(FAILED(res))
+          goto vcache;
```

Comment:
is there any reason why we need this label mess and don't just put the code here and exit early? Doesn't look like any of the other code is needed if this fails.

Reply:
Fair point, I can do that. I'm not a big fan of goto/labels anyway outside of assembly. Went with it more because of the "when in Rome" type of situation (we already have tons of "goto done" for error handling).